cpp/modmesh/buffer/small_vector.hpp

```diff
@@ -270,6 +270,66 @@ class small_vector
         m_head[m_size++] = value;
     }
 
+    T select_kth(size_t k)
+    {
+        iterator it = quick_select(begin(), end(), k);
+        return *it;
+    }
+
+    iterator choose_pivot(iterator left, iterator right)
+    {
+        iterator first = left;
+        iterator mid = left + (right - left) / 2;
+        iterator last = right - 1;
+        if (*mid < *first)
+            std::iter_swap(mid, first);
+        if (*last < *first)
+            std::iter_swap(last, first);
+        if (*last < *mid)
+            std::iter_swap(last, mid);
+        return mid;
+    }
+
+    iterator quick_select(iterator first, iterator last, size_t k)
```

Comment:
We need C++ unit tests for this.  Please create an issue to add the C++ unit tests for `small_vector::quick_select()`.